### PR TITLE
Create SharedAccordion component

### PR DESCRIPTION
**Pull request by jagodarybacka**
### What has been done
* Created component with collapsible element. 
* It allows to add custom elements to the header - always visible, and to the collapsible content part.
* Added simple rotation animation to the arrow button. Added showing/hiding animation to the content.

It is not used anywhere yet. You can test it using this snippet.
```javascript
    <SharedAccordion
      headerElement={<div>Header</div>}
      contentElement={
        <>
          <div>element</div>
          <div>element</div>
          <div>element</div>
          <div>element</div>
          <div>element</div>
          <div>element</div>
          <div>element</div>
        </>
      }
    />
```

### CODE REVIEW

File: ui/components/Shared/SharedAccordion.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import React, { ReactElement, useEffect, useRef, useState } from "react"
+import SharedIcon from "./SharedIcon"
+
+export default function SharedAccordion({
+  headerElement,
+  contentElement,
+  contentHeight,
+}: {
+  headerElement: ReactElement
+  contentElement: ReactElement
```
Comment (henryboldi) [inline ui/components/Shared/SharedAccordion.tsx:11]: I'm curious if an accordion component should accept multiple rows? This reads like a `SharedAccordionItem` to me

Comment (jagodarybacka) [reply ui/components/Shared/SharedAccordion.tsx:11]: what do you mean by "multiple rows"? like multiple headers that each is opening different content?

Comment (henryboldi) [reply ui/components/Shared/SharedAccordion.tsx:11]: > like multiple headers that each is opening different content?

Yes, typically accordion components allow multiple headers that each open different content.

Comment (jagodarybacka) [reply ui/components/Shared/SharedAccordion.tsx:11]: Right, but I think we can add this later as we won't need it in the nearest future.

**Review by henryboldi — APPROVED** [review]
Excellent! 👏👑

**Pull request merged**